Traceback (most recent call last):
  File "/home/hisaki/workspace/rl-semi/.venv/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/hisaki/workspace/rl-semi/.venv/lib/python3.8/site-packages/nbclient/client.py", line 1263, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/hisaki/workspace/rl-semi/.venv/lib/python3.8/site-packages/nbclient/util.py", line 85, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/hisaki/workspace/rl-semi/.venv/lib/python3.8/site-packages/nbclient/util.py", line 60, in just_run
    return loop.run_until_complete(coro)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/hisaki/workspace/rl-semi/.venv/lib/python3.8/site-packages/nbclient/client.py", line 701, in async_execute
    await self.async_execute_cell(
  File "/home/hisaki/workspace/rl-semi/.venv/lib/python3.8/site-packages/nbclient/client.py", line 1019, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/hisaki/workspace/rl-semi/.venv/lib/python3.8/site-packages/nbclient/client.py", line 913, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
%matplotlib inline
import matplotlib.pyplot as plt

env.reset()

plt.imshow(env.render(mode='rgb_array'))
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31merror[0m                                     Traceback (most recent call last)
Input [0;32mIn [2][0m, in [0;36m<cell line: 6>[0;34m()[0m
[1;32m      2[0m [38;5;28;01mimport[39;00m [38;5;21;01mmatplotlib[39;00m[38;5;21;01m.[39;00m[38;5;21;01mpyplot[39;00m [38;5;28;01mas[39;00m [38;5;21;01mplt[39;00m
[1;32m      4[0m env[38;5;241m.[39mreset()
[0;32m----> 6[0m plt[38;5;241m.[39mimshow([43menv[49m[38;5;241;43m.[39;49m[43mrender[49m[43m([49m[43mmode[49m[38;5;241;43m=[39;49m[38;5;124;43m'[39;49m[38;5;124;43mrgb_array[39;49m[38;5;124;43m'[39;49m[43m)[49m)

File [0;32m~/workspace/rl-semi/.venv/lib/python3.8/site-packages/gym/core.py:343[0m, in [0;36mWrapper.render[0;34m(self, **kwargs)[0m
[1;32m    341[0m [38;5;28;01mdef[39;00m [38;5;21mrender[39m([38;5;28mself[39m, [38;5;241m*[39m[38;5;241m*[39mkwargs):
[1;32m    342[0m     [38;5;124;03m"""Renders the environment with kwargs."""[39;00m
[0;32m--> 343[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43menv[49m[38;5;241;43m.[39;49m[43mrender[49m[43m([49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m

File [0;32m~/workspace/rl-semi/.venv/lib/python3.8/site-packages/gym/wrappers/order_enforcing.py:51[0m, in [0;36mOrderEnforcing.render[0;34m(self, **kwargs)[0m
[1;32m     46[0m [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m [38;5;28mself[39m[38;5;241m.[39m_disable_render_order_enforcing [38;5;129;01mand[39;00m [38;5;129;01mnot[39;00m [38;5;28mself[39m[38;5;241m.[39m_has_reset:
[1;32m     47[0m     [38;5;28;01mraise[39;00m ResetNeeded(
[1;32m     48[0m         [38;5;124m"[39m[38;5;124mCannot call `env.render()` before calling `env.reset()`, if this is a intended action, [39m[38;5;124m"[39m
[1;32m     49[0m         [38;5;124m"[39m[38;5;124mset `disable_render_order_enforcing=True` on the OrderEnforcer wrapper.[39m[38;5;124m"[39m
[1;32m     50[0m     )
[0;32m---> 51[0m [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43menv[49m[38;5;241;43m.[39;49m[43mrender[49m[43m([49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m

File [0;32m~/workspace/rl-semi/.venv/lib/python3.8/site-packages/gym/wrappers/env_checker.py:55[0m, in [0;36mPassiveEnvChecker.render[0;34m(self, **kwargs)[0m
[1;32m     53[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39mchecked_render [38;5;129;01mis[39;00m [38;5;28;01mFalse[39;00m:
[1;32m     54[0m     [38;5;28mself[39m[38;5;241m.[39mchecked_render [38;5;241m=[39m [38;5;28;01mTrue[39;00m
[0;32m---> 55[0m     [38;5;28;01mreturn[39;00m [43mpassive_env_render_check[49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43menv[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m
[1;32m     56[0m [38;5;28;01melse[39;00m:
[1;32m     57[0m     [38;5;28;01mreturn[39;00m [38;5;28mself[39m[38;5;241m.[39menv[38;5;241m.[39mrender([38;5;241m*[39m[38;5;241m*[39mkwargs)

File [0;32m~/workspace/rl-semi/.venv/lib/python3.8/site-packages/gym/utils/passive_env_checker.py:310[0m, in [0;36mpassive_env_render_check[0;34m(env, **kwargs)[0m
[1;32m    304[0m [38;5;28;01mif[39;00m render_fps [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[1;32m    305[0m     logger[38;5;241m.[39mwarn(
[1;32m    306[0m         [38;5;124m"[39m[38;5;124mNo render fps was declared in the environment (env.metadata[[39m[38;5;124m'[39m[38;5;124mrender_fps[39m[38;5;124m'[39m[38;5;124m] is None or not defined), [39m[38;5;124m"[39m
[1;32m    307[0m         [38;5;124m"[39m[38;5;124mrendering may occur at inconsistent fps[39m[38;5;124m"[39m
[1;32m    308[0m     )
[0;32m--> 310[0m [38;5;28;01mreturn[39;00m [43menv[49m[38;5;241;43m.[39;49m[43mrender[49m[43m([49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m

File [0;32m~/workspace/rl-semi/.venv/lib/python3.8/site-packages/gym/envs/toy_text/frozen_lake.py:248[0m, in [0;36mFrozenLakeEnv.render[0;34m(self, mode)[0m
[1;32m    246[0m     [38;5;28;01mreturn[39;00m [38;5;28mself[39m[38;5;241m.[39m_render_text(desc)
[1;32m    247[0m [38;5;28;01melse[39;00m:
[0;32m--> 248[0m     [38;5;28;01mreturn[39;00m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_render_gui[49m[43m([49m[43mdesc[49m[43m,[49m[43m [49m[43mmode[49m[43m)[49m

File [0;32m~/workspace/rl-semi/.venv/lib/python3.8/site-packages/gym/envs/toy_text/frozen_lake.py:260[0m, in [0;36mFrozenLakeEnv._render_gui[0;34m(self, desc, mode)[0m
[1;32m    258[0m [38;5;28;01mif[39;00m [38;5;28mself[39m[38;5;241m.[39mwindow_surface [38;5;129;01mis[39;00m [38;5;28;01mNone[39;00m:
[1;32m    259[0m     pygame[38;5;241m.[39minit()
[0;32m--> 260[0m     [43mpygame[49m[38;5;241;43m.[39;49m[43mdisplay[49m[38;5;241;43m.[39;49m[43minit[49m[43m([49m[43m)[49m
[1;32m    261[0m     pygame[38;5;241m.[39mdisplay[38;5;241m.[39mset_caption([38;5;124m"[39m[38;5;124mFrozen Lake[39m[38;5;124m"[39m)
[1;32m    262[0m     [38;5;28;01mif[39;00m mode [38;5;241m==[39m [38;5;124m"[39m[38;5;124mhuman[39m[38;5;124m"[39m:

[0;31merror[0m: No available video device
error: No available video device

